src/plugins/crypto/botan_operations.cpp

```diff
@@ -68,8 +71,9 @@ static int getKeyIvForEncryption (KeySet * config, Key * errorKey, Key * k, Symm
 			requiredKeyBytes, std::string (reinterpret_cast<const char *> (keyValue (msg)), keyGetValueSize (msg)), salt,
 			sizeof (salt), iterations);
 
-		*cKey = new SymmetricKey (derived.begin (), ELEKTRA_CRYPTO_BOTAN_KEYSIZE);
-		*cIv = new InitializationVector (derived.begin () + ELEKTRA_CRYPTO_BOTAN_KEYSIZE, ELEKTRA_CRYPTO_BOTAN_BLOCKSIZE);
+		cKey = unique_ptr<SymmetricKey> (new SymmetricKey (derived.begin (), ELEKTRA_CRYPTO_BOTAN_KEYSIZE));
+		cIv = unique_ptr<InitializationVector> (
+			new InitializationVector (derived.begin () + ELEKTRA_CRYPTO_BOTAN_KEYSIZE, ELEKTRA_CRYPTO_BOTAN_BLOCKSIZE));
 
 		delete pbkdf;
```

Comment:
You should use a unique_ptr for `get_pbkdf`, too (in the case of exceptions, the pbkdf will leak)